src/CartesianProduct.jl

```diff
@@ -549,3 +551,68 @@ function vertices_list(cpa::CartesianProductArray{N}
 
     return vlist
 end
+
+"""
+
+    block_indices(ca::CartesianProductArray{N}, vars::Vector{Int}) where {N}
+
+Return the dictionary of non-universal blocks (a key is the index of the block and a value is the index of the first variable in the block according to the whole set) of a (polytopic) Cartesian product of a finite
+number of sets.
+
+### Input
+
+- `ca` -- Cartesian product array
+- `vars` -- List of non free variables
+
+### Output
+
+A dictionary of block indices and indices of their first variables in the whole system.
+
+"""
+function block_indices(ca::CartesianProductArray{N}, vars::Vector{Int}) where {N}
```

Comment:
make `vars` optional, defaulting to the empty set? also, i don't understand why you restrict it to `Vector{Int}`, why not letting it not annotated, so other iterable structs can be used?

Reply:
on a second read, i would split this function on two smaller functions, one with one argument, and another one with no arguments. the behavior for `vars = empty` is generic (no need to dispatch on `CartesianProductArray`).

the behavior on the two argument function can also be used for other set types, so again i would not annotate the first argument to a `CartesianProductArray` (note that you can pass the input as `dim.(ca)`), instead use a vector (or abstractvector) of `Int`'s.